MSAL/src/MSALPublicClientApplication.m

```diff
@@ -727,31 +774,37 @@ - (void)acquireTokenSilentForScopes:(NSArray<NSString *> *)scopes
         completionBlock(result, error);
     };
 
-    params.unvalidatedAuthority = msidAuthority;
-    params.redirectUri = _redirectUri;
-    params.clientId = _clientId;
-    [params setClaims:claims error:&error];
-    if (error)
+    NSError *claimsError = nil;
+
+    // Set claims
+    if (![params setClaimsFromJSON:claims error:&claimsError])
     {
-        block(nil, error);
-        return;
+        block(nil, claimsError);
     }
 
-    error = nil;
-    MSALSilentRequest *request = [[MSALSilentRequest alloc] initWithParameters:params
-                                                                  forceRefresh:forceRefresh
-                                                                    tokenCache:self.tokenCache
-                                                              expirationBuffer:self.expirationBuffer
-                                                                         error:&error];
-    
-    if (!request)
+    MSIDDefaultTokenRequestProvider *tokenRequestProvider = [[MSIDDefaultTokenRequestProvider alloc] initWithOauthFactory:_oauth2Factory defaultAccessor:_tokenCache];
```

Comment:
_oauth2Factory & _tokenCache were created with default authority, but we are making request with authority provided in method parameters. Is it ok?

Reply:
Shouldn't happen, but addressed :)